New article "Simulate the Window Controls Overlay API without installing your PWA" (#3157)
MicrosoftDocs/edge-developer: microsoft-edge/devtools-guide-chromium/progressive-web-apps/simulate-window-controls-overlay.md

---
title: Simulate PWA Window Controls Overlay
description: Simulate PWA Window Controls Overlay.
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.service: microsoft-edge
ms.subservice: devtools
ms.date: 05/13/2024
---
# Simulate PWA Window Controls Overlay

> In DevTools, in the **Application** tool, you can simulate the Window Controls Overlay feature that Progressive Web Apps (PWAs) have, without having to install the PWA.  The **Application** tool has a **Window Controls Overlay** section under **Manifest**, to display a preview of how your Progressive Web App (PWA) handles window controls, without having to install the PWA and make the styling changes that are necessary to accommodate window controls:

Reviewer: ```suggestion
In DevTools, in the **Application** tool, you can simulate the Window Controls Overlay (WCO) feature that Progressive Web Apps (PWAs) have without having to install the PWA first and enable the WCO feature. By simulating the WCO feature from DevTools, you can test your title bar area CSS code changes more quickly.
```

Feel free to rephrase this, but it felt a little more intuitive to me like this.

![The 'Window Overlay Controls' section in the Application tool](./simulate-window-controls-overlay-images/pwa-window-controls.png)

> 1. In the manifest file, set the `display-override` field to `window-controls-overlay`:
> 
>    ```json
>    {
>      "display_override": ["window-controls-overlay"]
>    }
>    ```
> 
>    For more information, see [Enable the Window Controls Overlay in your app](../../progressive-web-apps-chromium/how-to/window-controls-overlay.md#enable-the-window-controls-overlay-in-your-app) in _Display content in the title bar_.

Reviewer: I would change this and start by assuming that developers have a PWA that uses the WCO feature already. How about:

```suggestion
1. Open a PWA that uses the WCO feature. For example, open the [1DIV demo app](https://microsoftedge.github.io/Demos/1DIV/dist/) in a new window or tab.

   For more information about using WCO in your PWA, see [Display content in the title bar](../../progressive-web-apps-chromium/how-to/window-controls-overlay.md).
```

Author: ```
1. Open a Progressive Web App (PWA) that uses the Window Controls Overlay (WCO) feature.
For example, open the [1DIV demo app](https://microsoftedge.github.io/Demos/1DIV/dist/) 
in a new window or tab.

   For more information about using WCO in your PWA, 
   see [Display content in the title bar](../../progressive-web-apps-chromium/how-to/window-controls-overlay.md).
```

1. In Microsoft Edge, select **Settings and more** > **More tools** > **Developer tools**.  Or, press **Ctrl+Shift+I** (Windows, Linux) or **Command+Option+I** (macOS).  DevTools opens.

1. On the **Activity Bar**, select  the **Application** tab.  If that tab isn't visible, click the **More tools** (![More tools icon](./simulate-window-controls-overlay-images/more-tools-icon.png)) button.

1. In the **Application** tool, expand **Manifest** in the upper left, and then select **Window Controls Overlay**.

> 1. In the **Window Controls Overlay** section on the right, select the checkbox **Emulate the Window Controls Overlay on** \<platform\>, and select a platform, such as Windows.
> 
> <!-- todo: then what?  what action, what result?
> 1. 
> -->

Reviewer: I think we need to describe the result of selecting the checkbox, and we need to add a screenshot showing the result:

```suggestion
1. In the **Window Controls Overlay** section on the right, select the checkbox **Emulate the Window Controls Overlay on** \<platform\>, and select a platform, such as Windows.

   The rendered web page now simulates the WCO feature:
   * An overlay that shows the window controls appears.
   * The values of the CSS `env(titlebar-area-left)`, `env(titlebar-area-top)`, `env(titlebar-area-width)`, and `env(titlebar-area-height)` variables are set to match the platform you selected.
```

Author: added:  todo: screenshot showing result

Use the Window Controls Overlay to have control over the title bar area that's currently displayed in standalone display mode.  Window Controls Overlay (WCO) is a set of features that work together to provide just the essential controls needed for the app window.  This layout frees up more space for the web content layer.
<!-- from https://learn.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/whats-new/pwa#window-controls-overlay-origin-trials -->

> Use the **Window Controls Overlay** section to emulate window controls for Windows, Linux, and macOS.

Reviewer: This belongs in a section earlier in the doc. Maybe a new H2 section after the intro? How about we structure the article like this:

```
# Simulate PWA Window Controls Overlay

Intro "In DevTools, in the **Application** tool, you can simulate the Window Controls Overlay (WCO) feature that Progressive Web Apps (PWAs)..."

## Using the WCO feature in your PWA

Quick reminder of what WCO is, and link to https://learn.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/window-controls-overlay

## Simulate WCO in DevTools

Your list of steps + screenshots.

---

> title: Simulate a Window Controls Overlay in a PWA's title bar area

Reviewer: ```suggestion
title: Simulate the Window Controls Overlay API without installing your PWA
```

Window Controls Overlay is really the name of the PWA feature (or API) here. So the phrase "a Window Controls Overlay" didn't make a lot of sense. The benefit of this new DevTools feature is to simulate WCO being used without having to first install the app locally. Feel free to propose different phrasing.

Author: globally updated title

> description: Display content in a PWA's title bar area by using the Window Controls Overlay API.

Reviewer: Proposal. This is really what the value proposition is.

```suggestion
description: Simulate enabling the Window Controls Overlay API in your Progressive Web App (PWA), to test your styles, without installing the PWA first.
```

Author: ```
description: Simulate enabling the Window Controls Overlay API in 
your Progressive Web App (PWA), to test your CSS styles, 
without having to install the PWA first.
```

author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.service: microsoft-edge
ms.subservice: devtools
ms.date: 05/21/2024
---
# Simulate a Window Controls Overlay in a PWA's title bar area

In DevTools, in the **Application** tool, you can simulate the Window Controls Overlay (WCO) feature that Progressive Web Apps (PWAs) have, without having to install the PWA first and enable the WCO feature.  By simulating the WCO feature from within DevTools, you can test CSS code changes for your title bar area more quickly than by re-installing the revised PWA.

![The 'Window Overlay Controls' section in the Application tool](./simulate-window-controls-overlay-images/pwa-window-controls.png)

<!-- ====================================================================== -->
## Using the WCO feature in your PWA

> In your Progressive Web App (PWA), use the Window Controls Overlay (WCO) to have control over the title bar area that's currently displayed in standalone display mode.  Window Controls Overlay is a set of features that work together to provide just the essential controls needed for the app window.  This layout frees up more space for the web content layer.
> 
> Use the **Window Controls Overlay** section to emulate window controls for Windows, Linux, and macOS.
> 
> See [Display content in the title bar area using Window Controls Overlay](../../progressive-web-apps-chromium/how-to/window-controls-overlay.md).

Reviewer: ```suggestion
In your Progressive Web App (PWA), use the Window Controls Overlay (WCO) to have control over the title bar area that's currently displayed in standalone display mode.  Window Controls Overlay is a set of features that work together to provide just the essential controls needed for the app window.  This layout frees up more space for the web content layer.

See [Display content in the title bar area using Window Controls Overlay](../../progressive-web-apps-chromium/how-to/window-controls-overlay.md).

Use the **Window Controls Overlay** section of the **Application** tool in DevTools to simulate the WCO feature for Windows, Linux, and macOS in your PWA without install the PWA first.
```

This ordering of the paragraphs makes more sense to me.

Author: without install**ING**

Author: ```
In your Progressive Web App (PWA), use the Window Controls Overlay (WCO) 
to have control over the title bar area that's currently displayed in standalone display mode.  
Window Controls Overlay is a set of features that work together 
to provide just the essential controls needed for the app window.  
This layout frees up more space for the web content layer.  
See [Display content in the title bar area using Window Controls Overlay](../../progressive-web-apps-chromium/how-to/window-controls-overlay.md).

To simulate the WCO feature for Windows, Linux, and macOS in your PWA 
without having to install the PWA first, 
use the **Window Controls Overlay** section of the **Application** tool in DevTools.
```

---
title: Display content in a PWA's title bar using Window Controls Overlay
description: Display content in a PWA's title bar using Window Controls Overlay.
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.service: microsoft-edge
ms.subservice: devtools
ms.date: 05/17/2024
---

> # Display content in a PWA's title bar using Window Controls Overlay

Author: best title of article?  describe the result as a static, simulated image of window controsl, per latest draft of article

Author: Latest titles: 
Simulate a Window Controls Overlay in a PWA's title bar area
Display content in the title bar area using Window Controls Overlay
